src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BytesKmerizer.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Iterator over successive Kmers from a byte[].
+ * Bytes are interpreted as ASCII characters.
+ * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
+ *
+ * Created by tsharpe on 12/18/15.
+ */
+public class BytesKmerizer implements Iterator<Kmer>
+{
+    public BytesKmerizer( final byte[] someSeq, final int someK )
+    {
+        seq = someSeq;
+        K = someK;
+        idx = 0;
+        kmer = new Kmer(K);
+        fill(0);
+    }
+
+    @Override
+    public boolean hasNext()
+    {
+        return kmer != null;
+    }
+
+    @Override
+    public Kmer next()
+    {
+        if ( kmer == null ) throw new NoSuchElementException("Kmerization sequence exhausted.");
+        final Kmer result = kmer;
+        fill(K-1);
+        return result;
+    }
+
+    public static Kmer toKmer( final byte[] seq )
+    {
+        final BytesKmerizer bk = new BytesKmerizer(seq,seq.length);
+        if ( !bk.hasNext() )
+            throw new IllegalArgumentException("Byte array contained illegal characters");
+        return bk.next();
+    }
+
+    private void fill( int validBaseCount )
+    {
+        Kmer kkk = kmer;
+        kmer = null;
+
+        final int len = seq.length;
+        while ( idx < len )
+        {
+            switch ( seq[idx] )
+            {
+                case 'A': kkk = kkk.successor(0,K); break;
+                case 'C': kkk = kkk.successor(1,K); break;
+                case 'G': kkk = kkk.successor(2,K); break;
+                case 'T': kkk = kkk.successor(3,K); break;
+                default: validBaseCount = -1;
+            }
+            idx += 1;
+
+            if ( ++validBaseCount == K )
+            {
+                kmer = kkk;
+                break;
+            }
+        }
+    }
+
+    private final byte[] seq;
+    private final int K;
```

Comment:
It looks a little weird to me to use a capital K for a variable name in Java style, so I'd switch to lower case unless you feel strongly about it.